Frameworks/Foundation/NSException.mm

```diff
@@ -216,7 +244,18 @@ - (HRESULT)_hresult {
         return [hresultValue unsignedIntValue];
     }
 
-    TraceWarning(L"NSException", L"Called -hresult on an NSException without an HRESULT!");
-    return E_UNEXPECTED;
+    return [[self class] _HRESULTForExceptionName:self.name];
+}
+
+- (void)_processException {
+    HRESULT hr = [self _hresult];
+    HSTRING_HEADER header;
+    HSTRING hsReason = nullptr;
+    
+    if(self.reason != nil) {
+        LPCWSTR reason = (LPCWSTR)([[self reason] cStringUsingEncoding:NSUnicodeStringEncoding]);
+        WindowsCreateStringReference(reason, wcslen(reason), &header, &hsReason);
+    }
+    RoOriginateError(hr, hsReason);
```

Comment:
assert sounds good enough probably. Just seems like an obvious place to be a little defensive in our programming. 

---
In reply to: [93384408](https://github.com/Microsoft/WinObjC/pull/1570#discussion_r93384408) [](ancestors = 93384408,93375861,93375624,93300590,93300439,93125265)